### PR TITLE
various cleanups, gitignore

### PR DESCRIPTION
**Pull request by matbesancon**
Added a gitignore with a Manifest.toml, which otherwise gets generated when running tests.
Made the include in the tests more robust with a joinpath

### CODE REVIEW

**Comment by fredrikekre** [comment]
Include is always relative the current file, so why are those changes needed,

**Comment by matbesancon** [comment]
usually yes, but testing stdlibs is a bit of an odd case (except if I missed an obvious workflow). To be sure the currently developed SparseArrays is being tested, I include the test file directly to test explicitly `Main.SparseArrays` which I have loaded before, in which case I always need to `cd test` before

**Comment by matbesancon** [comment]
@fredrikekre what's your opinion given the answer above?

File: test/higherorderfns.jl [hunk 1]
```diff
@@ -10,7 +10,7 @@ using Test
 using SparseArrays
 using LinearAlgebra
 using Random
-include("forbidproperties.jl")
+include(joinpath(@__DIR__, "forbidproperties.jl"))
```
Comment (vtjnash) [inline test/higherorderfns.jl:13]: This seems unnecessary complication, since `include` is already relative to `@__DIR__`

Comment (matbesancon) [reply test/higherorderfns.jl:13]: see this: https://github.com/JuliaSparse/SparseArrays.jl/pull/417#issuecomment-1646678186

Comment (SobhanMP) [reply test/higherorderfns.jl:13]: The workflow already checks that the correct version is loaded tho. See https://github.com/JuliaSparse/SparseArrays.jl/blob/main/.ci/test_and_change_uuid.jl

Comment (matbesancon) [reply test/higherorderfns.jl:13]: by workflow I meant local workflow, not github actions

**Comment by ViralBShah** [comment]
Should we merge?

**Comment by vtjnash** [comment]
I don't think it is clear why this is an improvement. It isn't how any other package does this.

**Comment by matbesancon** [comment]
for now it is not as any other package, since there is no good technique (to the best of my knowledge) to test locally

**Comment by SobhanMP** [comment]
but changing the local hash works fine?

**Comment by matbesancon** [comment]
> but changing the local hash works fine?

how do you do this?

**Comment by matbesancon** [comment]
in the meantime, it seems it's a problem on my side so I replaced it with https://github.com/JuliaSparse/SparseArrays.jl/pull/426

**Pull request closed without merging**